### PR TITLE
[sysrst_ctrl] added the markdown file

### PR DESCRIPTION
**Pull request by esshiu**
Signed-off-by: Eric Shiu <eshiu@google.com>

### CODE REVIEW

File: hw/ip/sysrst_ctrl/doc/_index.md [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+- Always on: uses the always-on power and clock domain
+- EC reset pulse duration control and stretching
+- Keyboard and button combo triggered action
+- AC_present can trigger interrupt 
+- Configuration registers can be set and locked until the next chip reset
+- Pin output override
+
+## Description
+
+The sysrst_ctrl logic is very simple. It looks up the configuration registers to decide how long the EC reset pulse duration and how long the keyboard debounce timer should be. Also what actions to take(eg. Interrupt, EC reset, GSC reset, disconnect the battery from the power tree).
```
Comment (msfschaffner) [inline hw/ip/sysrst_ctrl/doc/_index.md:25]: Can you go through this document and reformat it such that there is only one sentence per line? Thanks!

Also: nit: missing space before parenthesis `take(eg. ` -> `take (e.g.`

File: hw/ip/sysrst_ctrl/doc/_index.md [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+The sysrst_ctrl also controls an active low open drain reset I/O ec_rst_l (typically used for the embedded controller reset). This output is always asserted when the IP block is reset (allowing its use as a power-on reset) and remains asserted until released by software. A pulse width is configured for the reset signal. Trigger actions may cause the sysrst_ctrl to assert ec_rst_l for the specified width. If the ec_rst_l is asserted by some other agent in the system then the sysrst_ctrl will extend the assertion to ensure the minimum pulse width is generated.
+
+## Block Diagram
+
+![sysrst_ctrl Block Diagram](sysrst_ctrl_overview.svg)
+
+## Combo detection
+
+Let’s use the Power button+Esc+Refresh combo as an example
```
Comment (msfschaffner) [inline hw/ip/sysrst_ctrl/doc/_index.md:45]: nit: the `’` seems to be a non-ASCII character. 

can you go through this file and replace all of these apostrophes with `'`?

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 1 above)
Comment (msfschaffner) [reply hw/ip/sysrst_ctrl/doc/_index.md:25]: also, I would suggest to consistently use `e.g.` for `eg.` (there are more variations of this further below)

File: hw/ip/sysrst_ctrl/doc/_index.md [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+The SW defines three keys(eg. pwrb==0 && key0_in==0 && key1_in==0) as trigger in the configuration(com0_sel_ctl)
```
Comment (msfschaffner) [inline hw/ip/sysrst_ctrl/doc/_index.md:46]: missing spaces before parentheses `(` here and further below

File: hw/ip/sysrst_ctrl/doc/_index.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+The SW defines the duration(eg 10 seconds) that the above combo should be pressed in the configuration(com0_det_ctl)
+The SW defines the actions such as asserting ec_rst_l and raising interrupt in the configuration(com0_out_ctl)
+The SW defines the ec_rst_l pulse width in the configuration(ec_rst_ctl)
+The SW locks the configuration(no write is allowed) with cfg_wen_ctl
+sysrst_ctrl will detect the high(logic 1) to low(logic 0) transition on the combo and start the timer. If the timing condition is met(10 second), sysrst_ctrl will assert ec_rst_l,  sysrst_ctrl_intr and set the interrupt status register(intr_status) with the source
+The interrupt handler will read intr_status and write “logic 1” to clear the interrupt.
```
Comment (msfschaffner) [inline hw/ip/sysrst_ctrl/doc/_index.md:52]: `“”` are non ASCII characters (there are more further below)

can you replace those with `""`?

File: hw/ip/sysrst_ctrl/doc/_index.md [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+If the SW locks the configuration with cfg_wen_ctl, pin_allowed_ctl cannot be modified until GSC is reset
+When the system is up and running, the SW can modify pin_out_ctl and pin_out_value to enable or disable the feature. For example, to release ec_rst_l after GSC completes the reset, SW will program pin_out_ctl[1]=1’b0. The hardware will stop driving ec_rst_l=1’b0.
+The SW defines the keyboard polarity in key_invert_ctl. For inputs, it will be inverted before the combo logic. For outputs, it will be inverted after the pinout override logic.
+
+## EC and Power-on-reset
+
+GSC and EC will be reset together during power-on. When GSC is in reset, ec_rst_l will be asserted(active low).
+When GSC exits from reset, ec_rst_l should still be asserted. The power-on-reset value of pin_allowed_ctl[1] and pin_out_ctl[1] will guarantee ec_rst_l=1’b0 after GSC reset.
+The SW will release ec_rst_l explicitly by writing pin_out_ctl[1]=1’b0 after we enter the GSC-RW boot stage
+Now both GSC and EC complete the power-on-reset
```
Comment (msfschaffner) [inline hw/ip/sysrst_ctrl/doc/_index.md:89]: Missing periods at the end of the sentences.

**Review by msfschaffner — COMMENTED** [review]
Thanks @esshiu. A couple of format-related comments - see below.

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 1 above)
Comment (esshiu) [reply hw/ip/sysrst_ctrl/doc/_index.md:25]: Sure. One sentence per line. Use. e.g.

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 2 above)
Comment (esshiu) [reply hw/ip/sysrst_ctrl/doc/_index.md:45]: Sure.

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 3 above)
Comment (esshiu) [reply hw/ip/sysrst_ctrl/doc/_index.md:46]: Sure.

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 4 above)
Comment (esshiu) [reply hw/ip/sysrst_ctrl/doc/_index.md:52]: Sure.

File: hw/ip/sysrst_ctrl/doc/_index.md (hunk 5 above)
Comment (esshiu) [reply hw/ip/sysrst_ctrl/doc/_index.md:89]: Sure.

**Review by msfschaffner — APPROVED** [review]
Thanks @esshiu.

This probably needs another rebase onto master to get CI unstuck.

**Comment by esshiu** [comment]
Okay. rebase and push the branch.

**Pull request merged**